strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/MavenMetadataExpiredRepositoryPathHandler.java

```diff
@@ -0,0 +1,120 @@
+package org.carlspring.strongbox.providers.layout;
+
+import org.carlspring.strongbox.providers.io.RepositoryPath;
+import org.carlspring.strongbox.providers.io.RepositoryPathResolver;
+import org.carlspring.strongbox.providers.repository.ProxyRepositoryProvider;
+import org.carlspring.strongbox.providers.repository.RepositoryProvider;
+import org.carlspring.strongbox.providers.repository.RepositoryProviderRegistry;
+import org.carlspring.strongbox.storage.checksum.ChecksumCacheManager;
+import org.carlspring.strongbox.storage.metadata.MetadataHelper;
+import org.carlspring.strongbox.storage.repository.Repository;
+
+import javax.inject.Inject;
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import static org.carlspring.strongbox.providers.layout.MavenMetadataExpiredRepositoryPathHandler.Decision.I_DONT_KNOW;
+import static org.carlspring.strongbox.providers.layout.MavenMetadataExpiredRepositoryPathHandler.Decision.NO_LEAVE_IT;
+import static org.carlspring.strongbox.providers.layout.MavenMetadataExpiredRepositoryPathHandler.Decision.YES_FETCH;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+public class MavenMetadataExpiredRepositoryPathHandler
+        implements MavenExpiredRepositoryPathHandler
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(MavenMetadataExpiredRepositoryPathHandler.class);
+
+    @Inject
+    private RepositoryProviderRegistry repositoryProviderRegistry;
+
+    @Inject
+    private RepositoryPathResolver repositoryPathResolver;
+
+    @Inject
+    private ChecksumCacheManager checksumCacheManager;
```

Comment:
@fuss86 why you are using `ChecksumCacheManager` here? AFAIK it was only to validate checksum during artifact uploading. Also what was the goal to rework `ChecksumCacheManager`?

Reply:
@sbespalov , please review [this line](https://github.com/strongbox/strongbox/blob/master/strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/ArtifactManagementService.java#L238). It does nothing more than logging error in case of unmatched checksum cases.

I decided to use this cache for real case. I'm validating checksums using this cache.